test/fastmath.jl

```diff
@@ -202,3 +202,10 @@ let a = ones(2,2), b = ones(2,2)
     local c = 0
     @test @fastmath(c |= 1) == 1
 end
+
+struct LitPowTest end
+Base.literal_pow(::typeof(Base.FastMath.pow_fast), ::LitPowTest, ::Val{p}) where {p} = 1
```

Comment:
It doesn't look like the `literal_pow` mechanism itself is documented, let alone how it works with `@fastmath`. Should I try to document all of that in this PR or in another?